src/ex4/server/services/item_manager.js

```diff
@@ -1 +1,77 @@
-// The ItemManager should go here. Remember that you have to export it.
\ No newline at end of file
+// The ItemManager should go here. Remember that you have to export it.
+import PokemonClient from "../clients/pokemon_client.js";
+import fs from "fs";
+import envModule from "../../envModule.js";
+const cleanFile = { data: [] };
+class ItemManager {
+	constructor() {
+		this.pokemonClient = new PokemonClient();
+	}
+
+	getAllItems() {
+		let items = fs.readFileSync(envModule.jsonFilePath);
+		return JSON.parse(items);
+	}
+
+	async setAllItems(items) {
+		try {
+			fs.writeFileSync(envModule.jsonFilePath, JSON.stringify(items));
+		} catch (err) {
+			return err;
+		}
+	}
+
+	async addListItem(item) {
+		const items = this.getAllItems();
+		items.data = [...items.data, item];
+		await this.setAllItems(items);
+		return items;
+	}
+
+	removeItem(itemIndex) {
+		const items = this.getAllItems();
+		items.data = [...items.data.filter((item, i) => i !== parseInt(itemIndex))];
+		this.setAllItems(items);
+		return items;
+	}
+
+	async pokemonIdsHendeling(pokeID) {
+		const pokemon = await this.pokemonClient.getPokemon(pokeID);
+		return pokemon;
+	}
+
+	async pokemonFatching(pokemonIdsStr) {
```

Comment:
spelling mistake: pokemonFatching => pokemonFetching
same here, consider changing to something that sounds more like an action.. examples can be "fetchPokemons" or "fetchMultiplePokemon"